### PR TITLE
[NOREF] Simplify parse meta data logic and move to models

### PR DESCRIPTION
**Pull request by StevenWadeOddball**
## Description

@Peyton232 looked at how we are currently parsing meta data, and found it confusing to note where we have to match an enum to a type. We also determined that we could genericize the logic and move to a map. This makes the location of meta data and the enum in one close place, and only will require a new mapping when a new activity type is created.

## How to test this change

1.  Log into the app
2. Make new in app notifications
3.  Make sure to make multiples of the same type to validate that it shows a unique activity for each.

## PR Author Checklist

- [ ] I have provided a detailed description of the changes in this PR.
- [ ] I have provided clear instructions on how to test the changes in this PR.
- [ ] I have updated tests or written new tests as appropriate in this PR.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

### CODE REVIEW

File: pkg/models/activity.go [hunk 1]
```diff
@@ -42,3 +60,34 @@ func NewActivity(actorID uuid.UUID, entityID uuid.UUID, activityType ActivityTyp
 		ActivityType: activityType,
 	}
 }
+
+// ParseRawActivityMetaData conditionally parses the raw meta data JSON into the appropriate struct based on ActivityType
+// It returns a boolean indicating if parsing was successful and an error if any occurred during parsing
+func (a *Activity) ParseRawActivityMetaData() (bool, error) {
```
Comment (Peyton232) [inline pkg/models/activity.go:66]: I'm not quite seeing why we need a bool and error return type here, anytime there is an error we return false, and vice versa, so they seem coupled together so that your only returns are 

(false, err) or (true, nil)

**Review by Peyton232 — CHANGES_REQUESTED** [review]
Looks really good, passes functional test, also tested duplicates of notifications so all looks good. Only concern is why we are returning a success bool and err in the ParseRawActivityMetaData function, I may just be missing something though

File: pkg/models/activity.go (hunk 1 above)
Comment (StevenWadeOddball) [reply pkg/models/activity.go:66]: Yeah, I think that's fair! I'll work to adjust it

**Comment by Peyton232** [comment]
New discussion is being merged in right now, let's wait on this until that is shipped cause will need to add that to this map also

**Comment by StevenWadeOddball** [comment]
Thanks @Peyton232 ! For anyone taking a look, Peyton is going to wrap this up and merge once the other discussion notification is merged
